### PR TITLE
Clarifying single-user and token-based authentication on README.md issue #316

### PR DESCRIPTION
**Pull request by gcolliso**
Update to gnatsd/README.md clarifying single-user and token-based authentication (issue #316)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -451,54 +451,83 @@ This section describes how to secure the NATS server, including authentication,
...
 
-You can also enable single-user authentication and set the credentials in the server configuration file as follows:
+You can also enable token-based authentication and set the credentials in the server configuration file as follows:
 
 ```
 authorization {
-  user:     derek
-  password: T0pS3cr3t
-  timeout:  1
+  # You can generate the token using /util/mkpasswd.go
```
Comment (ColinSullivan1) [inline README.md]: Instead, I'd suggest something instead like
`# cleartext is supported but it is recommended you encrypt tokens with util/mkpasswd.go`

File: README.md [hunk 2]
```diff
@@ -451,54 +451,83 @@ This section describes how to secure the NATS server, including authentication,
...
-Or, if you chose to use a token:
+Clients can connect using:
+
+```
+nats://'S3Cr3T0k3n!'@localhost:4222
+```
+
+**Encrypting passwords and tokens**
+
+Passwords and tokens ideally should be be obfuscated with [bcrypt](#bcrypt).  Anywhere in a configuration file you store a password you should use the mkpasswd utility to encrypt the password or token and use that value instead.
```
Comment (ColinSullivan1) [inline README.md]: My mistake, substitute obfuscated with encrypted.

Comment (ColinSullivan1) [inline README.md]: Anywhere in a configuration file you store a password or token you should use the mkpasswd utility to encrypt and store that value instead.

File: README.md [hunk 3]
```diff
@@ -451,54 +451,83 @@ This section describes how to secure the NATS server, including authentication,
...
   # You can generate the token using /util/mkpasswd.go
-  token:   $2a$11$pBwUBpza8vdJ7tWZcP5GRO13qRgh4dwNn8g67k5i/41yIKBp.sHke
+  token:    $2a$11$UP3xizk94sWF9SHF/wkklOfBT9jphTGNrhZqz2OHoBdk9yO1kvErG
   timeout: 1
 }
 ```
 
->If you chose to use a token for client's authentication and generate the token by `/util/mkpasswd.go` then you must use the generated bcrypt hash as the token in server config, as written above, and the generated pass as the token in client configurations.
-
+If you want the mkpasswd util to generate a password or token for you, run it without the -p parameter.
```
Comment (derekcollison) [inline README.md]: utility spelled out here.

File: README.md [hunk 4]
```diff
@@ -451,54 +451,83 @@ This section describes how to secure the NATS server, including authentication,
...
 ```
 $ go run util/mkpasswd.go
-pass: D#6)e0ht^@61kU5!^!owrX // NATS client token
-bcrypt hash: $2a$11$bXz1Mi5xM.rRUnYRT0Vb2el6sSzVrqA0DJKdt.5Itj1C1K4HT9FDG // server authorization token
+pass: D#6)e0ht^@61kU5!^!owrX // Generated NATS client password (or token)
```
Comment (derekcollison) [inline README.md]: NATS client encrypted w/ Bcrypt password (or token)

**Comment by derekcollison** [comment]
I added a few minor comments and like @ColinSullivan1 suggestions as well. Otherwise LGTM.

**Pull request merged**